### PR TITLE
Build: Centralises binary naming conventions (build changes part 2)

### PR DESCRIPTION
**Pull request by am11**
* Delivers binary name, paths and download URL
  from lib/extensions.js.

* Allows user to set binary name as environment
  variable with `SASS_BINARY_NAME`.
  * Note: this will supersede default name.

* Allows user to set binary name as in package
  file `"nodeSassConfig": { "binary-name": "" }`.
  * Note: this will supersede default name and
    `SASS_BINARY_NAME` environment variable.

* Allows user to set binary name as parameter to
  invoke any node-sass script with
  `--binary-name` flag.
  * Note: this will supersede default name, name
    defined in package `nodeSassConfig` as well
    as the `SASS_BINARY_NAME` environment
    variable.

* This binary name will be used to construct the:
  * Binary path.
  * Binary download URL.
  * Upload URL.

* Allows user to set binary path as environment
  variable with `SASS_BINARY_PATH`.
  * Note: this will supersede default path.

* Allows user to set binary name as in package
  file `"nodeSassConfig": { "binary-path": "" }`.
  * Note: this will supersede default path and
    `SASS_BINARY_PATH` environment variable.

* Allows user to set binary path as parameter to
  invoke any node-sass script with
  `--binary-path` flag.
  * Note: this will supersede default path, path
    as well as the `SASS_BINARY_PATH` environment
    variable.

* This binary path will be used when:
  * Requiring node-sass package.
  * Downloading binary.
  * Uploading binary.

* Wraps all extensions in `process.sass`
  namespace.

Issue URL: #712.

//cc @nschonni, @xzyfer

### CODE REVIEW

File: test/api.js [hunk 1]
```diff
@@ -104,20 +104,16 @@ describe('api', function() {
...
-        var resolved = require.resolve('../lib');
-        delete require.cache[resolved];
-
         fs.renameSync(originalBin, renamedBin);
-        // try to re-require it
-        require('../lib');
+        process.sass.getBinaryPath(true);
       }, function(err) {
+        fs.renameSync(renamedBin, originalBin);
+        console.log(err);
```
Comment (nschonni) [inline test/api.js]: Was the `console` just for troubleshooting?

Comment (am11) [inline test/api.js]: It certainly was. Nice catch! :smile:

**Comment by nschonni** [comment]
@am11 what do you think about using a `nodeSassConfig` config element inside the `package.json` instead of the environment variables? EX: http://jshint.com/docs/

**Comment by am11** [comment]
@nschonni, I think it is a neat idea. Environment variables are bit over for this kind of a thing. But should we keep them anyway?

Precedence order (example):

- Process arguments `--binary-path ..`.
- `"nodeSassConfig": { "binary-path": ".." }`.
- Environment variable `SASS_BINARY_NAME`.

This way we will still not be the only (crazy) ones giving so many options, because we have node-jscs package for that! :wink:  (they have like 50 ways to provide settings)

**Comment by am11** [comment]
@nschonni, I have added the support for `nodeSassConfig.binaryName` and `nodeSassConfig.binaryPath`.

PTAL.

**Pull request merged**